### DISCUSSION TITLE
Rendering fields: Override Field Settings: checkbox selected

### DISCUSSION THREAD

**Discussion by mprofitlich** [verbb/formie#1550] [Q&A] [answered]
We are overwriting with Twig the checkboxes of a formie checkbox field (following the [documentation](https://verbb.io/craft-plugins/formie/docs/template-guides/rendering-fields#override-field-settings)). We now would like to make some of these checkboxes checked. We succeed in assigning a label and a value. 

Now: How to add the attribute checked to a checkbox? The following code does not assign a checked attribute. We did not find any option in the documentation; the line `checked: true` is just our guess.
```
{% do form.setFieldSettings('fieldname', {
    options:
        {
            label: "label text",
            value: "value text",
            checked: true
        },
        { … }
}) %}
```

**Comment by engram-design** [comment]
That would be a different step - you'll need to [populate](https://verbb.io/craft-plugins/formie/docs/template-guides/populating-forms#populating-forms) the submission with content.

**Reply by mprofitlich** [reply]
Thanks for your quick reply. I read the page you linked. I understand I can prepopulate values. I do not understand however how to set attributes such as checked. 

Could you help me with some code? 

What's the difference between overriding field settings and prepopulating?

**Reply by engram-design** [reply, marked as answer]
You can also use:
```twig
{% do form.setFieldSettings('fieldname', {
    options: [
        {
            label: "label text",
            value: "value text",
            isDefault: true
        },
        { … }
    ]
}) %}
```
But you can also do this with submission population.

```twig
{% do form.setFieldSettings('fieldname', {
    options: [
        {
            label: "label text",
            value: "value text",
        },
    ]
}) %}

{% do craft.formie.populateFormValues(form, { fieldname: ['value text'] }) %}
```

Field settings define how the field should render in the context of a form. That's before a submission exists, and the submission is where values for the form are stored. Think of it like setting the settings for a field in the CMS, and the entry is where you actually add the content. In the first example, I can use `isDefault` to set the default value of the Checkbox option. But I can also populate a submission with specific values, which will mark the corresponding option as checked. Either approach is fine.

**Reply by mprofitlich** [reply]
Thanks you for your explanation and the code examples. We succeeded thanks to your help.
